benchmarks/time/src/main/scala/strawman/collection/immutable/ArrayViewBenchmark.scala

```diff
@@ -0,0 +1,131 @@
+package strawman.collection.immutable
+
+import java.util.concurrent.TimeUnit
+
+import org.openjdk.jmh.annotations._
+import org.openjdk.jmh.infra.Blackhole
+import strawman.collection.{ArrayView, View}
+import scala.{Any, AnyRef, Int, Long, Unit, Array}
+import scala.Predef.intWrapper
+
+@BenchmarkMode(scala.Array(Mode.AverageTime))
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@Fork(1)
+@Warmup(iterations = 12)
+@Measurement(iterations = 12)
+@State(Scope.Benchmark)
+class ArrayViewBenchmark {
+
+  @Param(scala.Array("282", "73121", "7312102"))
+  var size: Int = _
+
+  @Param(scala.Array("39"))
+  var vLoSize: Int = _
+
+  var shortRangingFactor : Int = (size * 0.2).toInt
+
+  var v: View[Long] = _
+  var vLo: View[Long] = _
+
+  @Setup(Level.Trial)
+  def initData(): Unit = {
+
+    def fillArray(range: Int) = {
+      val array = new Array[Long](range)
+      var i = 0
+      while (i < range) {
+        array(i) = i.toLong
+        i += 1
+      }
+      array
+    }
+
+    v = ArrayView(fillArray(size))
+    vLo = ArrayView(fillArray(vLoSize))
+  }
+
+  @Benchmark
+  def sum (bh: Blackhole) = {
+    val ret : Long = v
+      .foldLeft(0L)(_+_)
+    bh.consume(ret)
+  }
+
+  @Benchmark
+  def sumOfSquares (bh: Blackhole) = {
+    val ret : Long = v
+      .map(d => d * d)
+      .foldLeft(0L)(_+_)
+    bh.consume(ret)
+  }
+
+  @Benchmark
+  def sumOfSquaresEven (bh: Blackhole) = {
+    val ret : Long = v
+      .filter(x  => x % 2L == 0L)
+      .map(x => x * x)
+      .foldLeft(0L)(_+_)
+    bh.consume(ret)
+  }
+
+  @Benchmark
+  def mapsMegamorphic (bh: Blackhole) = {
```

Comment:
This test isn't a good one, as it is a trivial candidate for fusion and might actually end up fused by the time the JIT compiler is done with it (and the compiler could do it too).  Unless it is actually a test for fusion, in which case the test name should be `def checkMapFusion`.